sdk/iot/Azure.Iot.Hub.Service/src/ConfigurationsClient.cs

```diff
@@ -0,0 +1,166 @@
+        public virtual Response<TwinConfiguration> CreateOrUpdateConfiguration(TwinConfiguration configuration, IfMatchPrecondition precondition = IfMatchPrecondition.IfMatch, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(configuration, nameof(configuration));
+            string ifMatchHeaderValue = IfMatchPreconditionExtensions.GetIfMatchHeaderValue(precondition, configuration.Etag);
+            return _configurationRestClient.CreateOrUpdate(configuration.Id, configuration, ifMatchHeaderValue, cancellationToken);
+        }
+
+        /// <summary>
+        /// Deletes a configuration on the IoT Hub for automatic device/module management
+        /// </summary>
+        /// <param name="configuration">Twin configuration to delete</param>
+        /// <param name="precondition">The condition on which to perform this operation. If no ETag is present on the twin configuration, then the condition must be equal to <see cref="IfMatchPrecondition.UnconditionalIfMatch"/>."/>.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The http response <see cref="Response{T}"/>.</returns>
+        public virtual Task<Response> DeleteConfigurationAsync(TwinConfiguration configuration, IfMatchPrecondition precondition = IfMatchPrecondition.IfMatch, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(configuration, nameof(configuration));
+            string ifMatchHeaderValue = IfMatchPreconditionExtensions.GetIfMatchHeaderValue(precondition, configuration.Etag);
+            return _configurationRestClient.DeleteAsync(configuration.Id, ifMatchHeaderValue, cancellationToken);
+        }
+
+        /// <summary>
+        /// Deletes a configuration on the IoT Hub for automatic device/module management
+        /// </summary>
+        /// <param name="configuration">Twin configuration to delete</param>
+        /// <param name="precondition">The condition on which to perform this operation. If no ETag is present on the twin configuration, then the condition must be equal to <see cref="IfMatchPrecondition.UnconditionalIfMatch"/>."/>.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The http response <see cref="Response{T}"/>.</returns>
+        public virtual Response DeleteConfiguration(TwinConfiguration configuration, IfMatchPrecondition precondition = IfMatchPrecondition.IfMatch, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(configuration, nameof(configuration));
+            string ifMatchHeaderValue = IfMatchPreconditionExtensions.GetIfMatchHeaderValue(precondition, configuration.Etag);
+            return _configurationRestClient.Delete(configuration.Id, ifMatchHeaderValue, cancellationToken);
+        }
+
+        /// <summary>
```

Comment:
if the count is left as null, how many does the service send back ? is there a cap?

Reply:
I think that is worth mentioning in the comments.

Reply:
I do not know, I'll add this question to our ongoing thread